code/modules/antagonists/changeling/powers/fakedeath.dm

```diff
@@ -51,7 +52,13 @@
 /// Can be called mid-revival if the process is being cancelled
 /datum/action/changeling/fakedeath/proc/disable_revive(mob/living/changeling)
 	if(revive_ready)
-		chemical_cost = 15
+		// added a spawn(1) before we return chemical_cost to 15,
+		// as this verb is callstacked up to sting_action, whose return value is checked
+		// by try_to_sting. try_to_sting doesn't deduct chemicals until AFTER the whole
+		// revival is resolved, so we're delaying it with the spawn to stop the double
+		// chemical deduction. see line 74
+		spawn(1)
+			chemical_cost = 15
```

Comment:
please don't invoke `spawn(1)` directly in code like this, we consider these sections of the code to be more "user-facing" (as in it's reasonably expected for any new contributor to start crawling in here and toying with stuff and being utterly confused despite the explanation). if you really need to having a blocking sleep or something I would rather you invoke `sleep(0.1 SECONDS)` instead, but i feel like there should be a better answer to this anyways.

Reply:
```txt
	// This is one of very few spawn()s left in our codebase
	// Yes it needs to be like this
	// No, you're not allowed to use spawn() yourself, unless you're making something that needs to work outside timers
	// Don't be a dumbass, I will always be watching
```

pretty much the only uses of spawn() in the code nowadays are stuff that is placed direclty on `World` or `ASYNC` stuff. you will give poor lemoninthedark a big frowny face if this makes it into production code and i don't know how many more times his muscles can make that face before it gets permanently fused into that position.

Reply:
i'll need to refactor the code if i'm not able to use the spawn() there. the code to reset the chemical cost is called right in the middle of the revive() proc. i'll make another PR, as I'll have to go at this from the ground up in that case.

Reply:
in any case i accidentally pushed another entirely separate PR into this one so closing it is probably best, haha.

Reply:
Yeah I think a deep cleaning of how this works is warranted if that's the case. Hrm